Site: Add blog post for iceberg-rust 0.10.0 (#17032)
apache/iceberg: site/docs/blog/posts/2026-07-21-iceberg-rust-0.10.0-release.md

### Storage Improvements

#### New Storage Backend - HuggingFace Hub

A new [HuggingFace Hub storage backend](https://github.com/apache/iceberg-rust/pull/2375) was contributed, enabling direct access to Iceberg tables hosted on HuggingFace datasets.

#### Other improvements

- **[`delete_stream` on Storage trait](https://github.com/apache/iceberg-rust/pull/2216)**: Batch delete operations for efficient cleanup
- **[OpenDAL resolving storage](https://github.com/apache/iceberg-rust/pull/2231)**: Automatically resolves the appropriate storage backend based on file path scheme
- **[Timeout layer for OpenDAL](https://github.com/apache/iceberg-rust/pull/2455)**: Added `TimeoutLayer` inside `RetryLayer` to prevent indefinite hangs
- **[S3 virtual-host-style addressing](https://github.com/apache/iceberg-rust/pull/2330)**: Default to virtual-host-style S3 addressing for broader compatibility

> ## Bug Fixes

Reviewer: https://github.com/apache/iceberg-rust/pull/2351 was a correctness issue also if you wanted to include that?

Notable correctness fixes in this release:

- **[INT96 timestamp values](https://github.com/apache/iceberg-rust/pull/2301)**: Fixed incorrect Parquet INT96 timestamp interpretation
- **[Nested type column indices](https://github.com/apache/iceberg-rust/pull/2307)**: Fixed `build_fallback_field_id_map` producing incorrect indices for schemas with nested types
- **[Delete-only manifests in FastAppend](https://github.com/apache/iceberg-rust/pull/2545)**: Preserved delete-only manifests that were incorrectly dropped
- **[DeleteFileIndex lost wakeup](https://github.com/apache/iceberg-rust/pull/2696)**: Fixed a hang caused by a lost-wakeup race in `get_deletes_for_data_file`
- **[Row group byte range filtering](https://github.com/apache/iceberg-rust/pull/2615)**: Fixed row duplication for sub-row-group file splits
- **[Cross-engine decimal compatibility](https://github.com/apache/iceberg-rust/pull/2538)**: Added space in decimal type serialization for compatibility with other engines

> - **[Puffin magic number validation](https://github.com/apache/iceberg-rust/pull/2416)**: Added missing validation for magic number at the start of a Puffin file

Reviewer 2: nit: In my opinion, adding validation is usually not a bug fix. However, there's no need to remove it.

[changelog]: https://github.com/apache/iceberg-rust/releases/tag/v0.10.0

`iceberg-rust` is a native Rust implementation of the Apache Iceberg specification, providing high-performance libraries for reading, writing, and managing Iceberg tables in Rust applications and through Python bindings (`pyiceberg-core`).

<!-- more -->

## Release Highlights

There have been plenty of contributions across the `iceberg-rust` library; here's a highlight of a few.

### Schema Evolution

This release adds the [`update_schema` action](https://github.com/apache/iceberg-rust/pull/2120), enabling column addition and removal.

### Snapshot Expiration

A new `ExpireSnapshotsAction` ([#2591](https://github.com/apache/iceberg-rust/pull/2591), [#2664](https://github.com/apache/iceberg-rust/pull/2664), [#2667](https://github.com/apache/iceberg-rust/pull/2667)) provides a high-level API for expiring old snapshots, respecting `history.expire.*` table properties and cleaning up associated statistics file metadata.

### Configurable Tokio runtime

> The [`Runtime` trait](https://github.com/apache/iceberg-rust/pull/2308) can now be provided when initializing catalogs.

Reviewer 3: ```suggestion
A [custom tokio `Runtime`](https://github.com/apache/iceberg-rust/pull/2308) can now be provided when initializing catalogs.
```
